src/parser/clauses/unwind.c

```diff
@@ -0,0 +1,33 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Apache License, Version 2.0,
+* modified with the Commons Clause restriction.
+*/
+
+#include "unwind.h"
+#include "../ast_arithmetic_expression.h"
+
+AST_UnwindNode* New_AST_UnwindNode(Vector *expressions, char *alias) {
+    AST_UnwindNode *unwindNode = malloc(sizeof(AST_UnwindNode));
+    unwindNode->expressions = expressions;
+    unwindNode->alias = alias;
+    return unwindNode;
+}
+
+void UnwindClause_ReferredEntities(const AST_UnwindNode *unwindNode, TrieMap *referred_entities) {
+    if(!unwindNode) return;
+    TrieMap_Add(referred_entities, unwindNode->alias, strlen(unwindNode->alias), NULL, TrieMap_DONT_CARE_REPLACE);
```

Comment:
This would be another place where we could check the validity of entities in `unwindNode->expressions`, though it does have the slight complication that UNWIND introduces a new alias while assuming that others exist.